v3/lgr_physics.cpp

```diff
@@ -221,6 +222,87 @@ HPC_NOINLINE inline void neo_Hookean(input const& in, state& s, material_index c
   hpc::for_each(hpc::device_policy(), s.element_sets[material], functor);
 }
 
+HPC_NOINLINE inline void hyper_ep_update(input const& in, state& s, material_index const material) {
+  // Constant state
+  auto const points_to_dt = s.dt;
+  auto const points_to_F_total = s.F_total.cbegin();
+  auto const points_to_temp = s.temp.cbegin();
+
+  // Variables to be updated
+  auto points_to_Fp_total = s.Fp_total.begin();
+  auto points_to_sigma = s.sigma.begin();
+  auto points_to_ep = s.ep.begin();
+  auto points_to_ep_dot = s.ep_dot.begin();
+  auto points_to_dp = s.dp.begin();
+  auto points_to_localized = s.localized.begin();
+
+  // Elastic parameters to be updated
+  auto const points_to_K = s.K.begin();
+  auto const points_to_G = s.G.begin();
+
```

Comment:
@ibaned it might be nice to allow material models to have a single `ui` array.  eg, `auto props = in.material_params[material];`

Reply:
where `material_params` is an array that is filled with the user input.

Reply:
I do like that idea, would be okay with that refactor.